mock_server/closer.go

```diff
@@ -0,0 +1,34 @@
+//go:build go1.18
```

Comment:
This is all taken from azcore internal mock server implementation that they use for testing. I could not import it in azcopy since its internal. @vibhansa-msft can we ask them if they can make it public for us?